billing/checkout/service.go

```diff
@@ -728,6 +728,47 @@ func (s *Service) CreateSessionForPaymentMethod(ctx context.Context, ch Checkout
 	})
 }
 
+func (s *Service) CreateSessionForCustomerPortal(ctx context.Context, ch Checkout) (Checkout, error) {
+	// get billing
+	billingCustomer, err := s.customerService.GetByID(ctx, ch.CustomerID)
+	if err != nil {
+		return Checkout{}, err
+	}
+
+	checkoutID := uuid.New().String()
+
+	sessionParams := &stripe.BillingPortalSessionParams{
+		Params: stripe.Params{
+			Context: ctx,
+		},
+		Customer: stripe.String(billingCustomer.ProviderID),
+	}
+
+	if ch.CancelUrl != "" {
+		sessionParams.ReturnURL = stripe.String(ch.CancelUrl)
```

Comment:
does that mean setting return url is optional? Also how about success URL?

Reply:
billing session params only need return url. which is optional.
https://github.com/stripe/stripe-go/blob/v72.122.0/billingportal_session.go#L40